vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -81,31 +82,21 @@ public ITextBlock createTextBlock(String slideId, String text, Integer contentOr
         ITextBlock textBlock = textBlockFactory.createTextBlock(slide, text);
         textBlock.setContentOrder(contentOrder);
         textBlock = textBlockRepo.save((TextBlock) textBlock);
-        
         return textBlock;
     }
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see edu.asu.diging.vspace.core.services.impl.IContentBlockManager#
-     * createImageBlock(java.lang.String, java.util.Arrays, java.lang.String)
-     */
-    @Override
-    public CreationReturnValue createImageBlock(String slideId, byte[] image, String filename, Integer contentOrder) throws ImageCouldNotBeStoredException {
-        IVSImage slideContentImage = null;
-        ISlide slide = slideManager.getSlide(slideId);
+    private IVSImage saveImage(byte[] image, IVSImage slideContentImage, String filename) {
         if (image != null && image.length > 0) {
             Tika tika = new Tika();
             String contentType = tika.detect(image);
-
             slideContentImage = imageFactory.createImage(filename, contentType);
             slideContentImage = imageRepo.save((VSImage) slideContentImage);
         }
+        return slideContentImage;
+    }
 
-        CreationReturnValue returnValue = new CreationReturnValue();
-        returnValue.setErrorMsgs(new ArrayList<>());
-
+    private IVSImage handleNullContent(byte[] image, IVSImage slideContentImage, String filename)
```

Comment:
you actually handle the case that the content isn't null, don't you?

Reply:
also, since you are passing ```slideContentImage``` there is no need to return it as well since you just modify it's attributes.